module/VuFind/src/VuFind/Controller/CommentsController.php

```diff
@@ -0,0 +1,123 @@
+ * @link     https://vufind.org Main Page
+ */
+class CommentsController extends AbstractBase
+{
+    /**
+     * Get all comments for the logged in user
+     *
+     * @return View
+     */
+    public function userCommentsAction()
+    {
+        $user = $this->getUser();
+        if (!$user) {
+            return $this->forceLogin();
+        }
+        if (
+            !$this->commentsEnabled()
+            && !$this->getService(\VuFind\Config\AccountCapabilities::class)->getRatingSetting()
+        ) {
+            throw new ForbiddenException('Comments and ratings disabled.');
+        }
+        $limit = 50;
+        $page = $this->params()->fromQuery('page', 1);
+        $service = $this->getDbService(\VuFind\Db\Service\CommentsServiceInterface::class);
+        $comments = $service->getCommentsAndRatingsByUserId(
+            $user->id,
+            $limit,
+            $page,
+            $this->commentsEnabled(),
+            $this->getService(\VuFind\Config\AccountCapabilities::class)->getRatingSetting()
+        );
+        $recordLoader = $this->serviceLocator->get(\VuFind\Record\Loader::class);
+        $ids = [];
+        foreach ($comments as $comment) {
+            $ids[] = $comment['source'] . '|' . $comment['record_id'];
+        }
+        $records = $recordLoader->loadBatch($ids, true);
+        foreach ($comments as $i => $c) {
+            $c['recordTitle'] = $records[$i]->getTitle() ?? '';
+        }
+        return $this->createViewModel(['comments' => $comments, 'params' => $this->params()->fromQuery()]);
+    }
+
+    /**
+     * Delete given comments by the logged in user
+     *
+     * @return View
+     */
+    public function deleteCommentsAction()
+    {
+        $user = $this->getUser();
+        if (!$user) {
+            return $this->forceLogin();
+        }
+        if ($this->formWasSubmitted(['deleteSelectedcomment', 'deleteSelectedrating'])) {
+            $csrf = $this->getService(CsrfInterface::class);
+            if (!$csrf->isValid($this->getRequest()->getPost()->get('csrf'))) {
+                throw new \VuFind\Exception\BadRequest(
+                    'error_inconsistent_parameters'
+                );
+            }
+        }
+
+        if (!empty($comments = $this->params()->fromPost('deleteSelectedcomment', []))) {
+            $commentsService = $this->getDbService(\VuFind\Db\Service\CommentsServiceInterface::class);
+            $commentsService->deleteByIdsAndUserId($comments, $user->getId());
+        }
+
+        if (
+            !empty($ratings = $this->params()->fromPost('deleteSelectedrating', []))
+            && $this->getService(\VuFind\Config\AccountCapabilities::class)->isRatingRemovalAllowed()
+        ) {
+            $ratingsService = $this->getDbService(\VuFind\Db\Service\RatingsServiceInterface::class);
+            $ratingsService->deleteByIdsAndUserId($ratings, $user->getId());
+        }
+
+        return $this->redirect()->toRoute(
+            'default',
```

Comment:
I think it would be simpler to redirect to the `comments-usercomments` route here instead of using `default` with parameters.